src/careamics/models/lvae/noise_models.py

```diff
@@ -1,135 +1,76 @@
-                    nmodel1 = GaussianMixtureNoiseModel(params=np.load(nmodel_fpath))
-                    nmodel2 = None
-                    nmodels = [nmodel1, nmodel2]
-                else:
-                    raise ValueError(f"Invalid denoise_channel: {denoise_channel}")
-                # TODO WTF are input and ch1 the same ?
-        # muSplit -> two outputs -> two noise models
-        elif noise_model_type == "gmm":
-            print(f"Noise model Ch1: {noise_model_ch1_fpath}")
-            print(f"Noise model Ch2: {noise_model_ch2_fpath}")
-
-            nmodel1 = GaussianMixtureNoiseModel(params=np.load(noise_model_ch1_fpath))
-            nmodel2 = GaussianMixtureNoiseModel(params=np.load(noise_model_ch2_fpath))
-
-            nmodels = [nmodel1, nmodel2]
-
-            # if 'noise_model_ch3_fpath' in config.model:
-            #     print(f'Noise model Ch3: {config.model.noise_model_ch3_fpath}')
-            #     nmodel3 = GaussianMixtureNoiseModel(params=np.load(config.model.noise_model_ch3_fpath))
-            #     nmodels = [nmodel1, nmodel2, nmodel3]
-            # else:
-            #     nmodels = [nmodel1, nmodel2]
-        else:
-            raise ValueError(f"Invalid noise_model_type: {noise_model_type}")
+        return MultiChannelNoiseModel(noise_models)
+    # TODO should config and paths be mutually exclusive? What you mean, Igor?
+    return None
 
-        if noise_model_learnable:
-            for nmodel in nmodels:
-                if nmodel is not None:
-                    nmodel.make_learnable()
 
-        return DisentNoiseModel(*nmodels)
-    return None
+class MultiChannelNoiseModel(nn.Module):
+    def __init__(self, nmodels: list[GaussianMixtureNoiseModel]):
+        """Constructor.
 
+        To handle noise models and the relative likelihood computation for multiple
+        output channels (e.g., muSplit, denoiseSplit).
 
-class DisentNoiseModel(nn.Module):
-    # TODO this class is a mess(this is an autogenerated btw). But really wtf?!?!?!
-    def __init__(self, *nmodels):
-        """
-        Constructor.
+        This class:
+        - receives as input a variable number of noise models, one for each channel.
+        - computes the likelihood of observations given signals for each channel.
+        - returns the concatenation of these likelihoods.
 
-        This class receives as input a variable number of noise models, each one corresponding to a channel.
+        Parameters
+        ----------
+        nmodels : list[GaussianMixtureNoiseModel]
+            List of noise models, one for each output channel.
         """
         super().__init__()
-        # self.nmodels = nmodels
         for i, nmodel in enumerate(nmodels):
             if nmodel is not None:
-                self.add_module(f"nmodel_{i}", nmodel)
+                self.add_module(
+                    f"nmodel_{i}", nmodel
+                )  # TODO: wouldn't be easier to use a list?
```

Comment:
In reply to TODO: it might be easier, yes, but you have to use [`torch.nn.ModuleList`](https://pytorch.org/docs/stable/generated/torch.nn.ModuleList.html)